grafana/plugin-tools
# Docs: Plugin-E2E: Docs improvements

**What this PR does / why we need it**:

@mckn have read through the `Get started` docs, and he gave me the following feedback: 
* We should have `You will learn` summary at the top of the page (similar to the [Playwright docs](https://playwright.dev/docs/intro#introduction))
* Move part about `.gitignore` from Installing Playwright section to Configure Playwright
* Move part about removing files from Installing Playwright section to Writing tests

All these bullet point should be addressed in this PR.

Furthermore, he suggested to drop the idea of having examples based on a scaffolded backend data source, and instead simply test that auth (the config) works. So instead of having plugin type specific tests in the Get started guide, we provide links to the following guides: 
* How to test a panel plugin
* How to test a data source plugin
* How to test an app plugin

I like this idea. However, since we don't have docs for any of these plugin types yet, I think we should keep the data source specific example for now as it, in only a few lines of code, showcases what plugin-e2e is capable of. Once we have published docs for how to test the different plugin types, we can update the Get started guide again any make it plugin type agnostic. Thoughts @mckn? 

Aside from that, this PR includes the following changes:
* Adds the `introduction` page to the sidebar menu. This content on this page is information, and I'm concerned people would miss out on it if it's not present in the sidebar menu.
* Align content in the `authentication.md` article with the latest changes in the get-started guide. 
* Remove the code formatting from all headers as it looks weird
 
**Which issue(s) this PR fixes**:



Fixes #

**Special notes for your reviewer**:
